client/lib/signup/cart.js

```diff
@@ -11,7 +11,33 @@ var wpcom = require( 'lib/wp' ),
 	cartValues = require( 'lib/cart-values' ),
 	cartItems = cartValues.cartItems;
 
+function addProductsToCart( cart, newCartItems ) {
+	forEach( newCartItems, function( cartItem ) {
+		cartItem.extra = Object.assign( cartItem.extra || {}, {
+			context: 'signup'
+		} );
+		const addFunction = cartItems.add( cartItem );
+
+		cart = cartValues.fillInAllCartItemAttributes( addFunction( cart ), productsList.get() );
+	} );
+
+	return cart;
+}
+
 module.exports = {
+	createCart: function( cartKey, newCartItems, callback ) {
+		let newCart = {
+			cart_key: cartKey,
+			products: [],
+			temporary: false,
+		};
+
+		newCart = addProductsToCart( newCart, newCartItems );
+
+		wpcom.undocumented().cart( cartKey, 'POST', newCart, function( postError ) {
```

Comment:
@drewblaisdell I'm not sure I got what you meant, `callback` here doesn't accept `providedDependencies`
That function is called like that:
```js
SignupCart.createCart( cartKey, [ domainItem ], error => callback( error, providedDependencies ) );
```

So `callback` inside `createCart` is of the shape `error => {}`, even if we would pass additional parameters to it, like `callback( error, data, drew1, drew2 )`, that wouldn't change anything regarding `providedDependencies` as it's captured by the closure of whatever is calling `createCart` ( in that case I copied it from [createSiteOrDomain step](https://github.com/Automattic/wp-calypso/pull/11530/files#diff-160d380c201b4db2d947c495e877a98aR41) )